golang/vaas/pkg/vaas/vaas.go

```diff
@@ -623,36 +635,26 @@ func (v *vaas) runPingTicker(ctx context.Context) {
 				}
 				return
 			}
-		case <-ctx.Done():
+		case <-v.termChan:
 			return
 		}
 	}
 }
 
-func (v *vaas) listenWebSocket(ctx context.Context) chan error {
-	termChan := make(chan error, 1)
-	listenCtx, listenCancel := context.WithCancel(ctx)
+func (v *vaas) listenWebSocket() chan error {
+	errorChan := make(chan error, 1)
+	v.termChan = make(chan bool)
 
-	go func() {
-		<-listenCtx.Done()
-		if err := v.websocketConnection.Close(); err != nil && v.options.EnableLogs {
-			v.logger.Printf("Failed to close web socket: %v", err)
-		}
-		close(termChan)
-	}()
+	go v.runPingTicker()
+	go v.readWebSocket(errorChan)
```

Comment:
I would prefer to move the go routine inside `readWebSocket` for two reasons:
1. If you do it like this you always have to start the function in a go routing, otherwise the call blocks and you can not read from the channel
2. Not sure if this is an "offical idiom" but for me the one who creates the channel should always be the one who closes it, too. When you pass a channel to a function and the function closes the channel it is some unexpected behaviour and can cause crashes.

```suggestion
	errorChan := v.readWebSocket()
```

Reply:
regarding `2.`: So you are basically saying go routins should be anonymous?

Reply:
I tried to alter as little as possible when it comes to design choices. I would leave this up to @unglaublicherdude if that's ok. Feel free to alter this PR if you think this is a sensible suggestion.

Reply:
I don't have a heavy opinion here, because we don't really use this SDK in bigger projects. So while you are the ones actually using the SDK I will let you make the decision. But that's just me, maybe @GermanCoding @pstadermann @doxthree or @lennartdohmann have a different opinion on that.

Reply:
I just went ahead and implemented the requested changes in order to avoid further discussion and get this merged quickly. ;)

Reply:
@cavus700 @rageagainsthepc do you agree on this PR now? If so, I don't mind merging it.

Reply:
The code itself looks good to me but in my opinion these changes still require a v2 and I would not merge it without. 
Since this is not my project and I don't know all stakeholders the decision is up to you.